config.c

```diff
@@ -21,10 +21,34 @@
 #define NUMBER_OF_SHADERS 5
 
 #ifdef _MSC_VER
+#include "Shlwapi.h"
 #include "Windows.h"
+#define TEXTFILE 256
+#define IDR_CONFIG_FILE 101
+#define IDR_BAR_SPECTRUM_SHADER 102
+#define IDR_NORTHERN_LIGHTS_SHADER 103
+#define IDR_PASS_THROUGH_SHADER 104
+#define IDR_SPECTROGRAM_SHADER 105
+#define IDR_WINAMP_LINE_STYLE_SPECTRUM_SHADER 106
 #define PATH_MAX 260
 #define PACKAGE "cava"
 #define _CRT_SECURE_NO_WARNINGS 1
+
+static void LoadFileInResource(int name, int type, DWORD *size, const char **data) {
+    HMODULE handle = GetModuleHandle(NULL);
+    HRSRC rc = FindResource(handle, MAKEINTRESOURCE(name), MAKEINTRESOURCE(type));
+    HGLOBAL rcData = LoadResource(handle, rc);
+    if (size) {
+        *size = SizeofResource(handle, rc);
+    }
+    if (data) {
+        *data = (const char *)(LockResource(rcData));
+    }
+}
+
+int default_shader_data[NUMBER_OF_SHADERS] = {IDR_BAR_SPECTRUM_SHADER, IDR_NORTHERN_LIGHTS_SHADER,
```

Comment:
the order here has to match the order on line 72, the `default_shader_name` array.

Reply:
ah okay. fixed!